### PR TITLE
feat: upgrade spec format to 7-section standard in AGENTS.md

### LINKED ISSUE #119: feat: upgrade spec format in AGENTS.md detailing stage
## Problem

\`AGENTS.md\` (root) has no spec format section — agent writes issue body
in free prose. \`templates/AGENTS.md\` has spec format but missing 4 sections:
out of scope, NFRs, edge cases, success metrics.

Result: Jules receives issues with only happy path → over-engineers scope,
misses edge cases, skips NFRs.

## Sprint Goal

| Dimension | Before | Target |
|---|---|---|
| Out of scope | 1/10 | 10/10 |
| Given/When/Then | 3/10 | 10/10 |
| NFRs com número | 1/10 | 10/10 |
| Success metrics em features | 2/10 | 10/10 |
| Edge cases sistemáticos | 3/10 | 10/10 |
| AC em features/enhancements | 2/10 | 10/10 |

Measured on: next 5 non-bug issues created after this PR merges.

## Solution

Add/replace `## Spec Format` section in both AGENTS.md files with the
7-section mandatory format below. Remove user story wrapper ("As [user]...") —
problem statement is more precise for agents.

### Spec format to enforce

```
## Problem
[1-3 sentences. What fails. Who affected. Measured impact.]

## Sprint Goal / Success Metrics
| Metric | Baseline | Target | When |
|--------|----------|--------|------|

## Solution
[Behavioral description of what is built. No implementation details.]

## Acceptance Criteria
**AC1 — [name]**
- Given [precondition]
- When [action]
- Then [outcome]

## Edge Cases
- EC1: [condition] → [expected behavior]

## Out of Scope
- [item] — reason

## Non-Functional Requirements
- Performance: [metric with number]
- Security: [constraint]
- Reliability: [constraint]
> Omit NFRs only for pure docs/markdown issues with zero runtime behavior.

## Files to Modify
- `path/to/file` — what changes
```

## Acceptance Criteria

**AC1 — Root AGENTS.md has 7-section spec format**
- Given detailing agent reads root `AGENTS.md`
- When agent writes issue body during detailing
- Then output contains all 7 sections: Problem, Sprint Goal, Solution, ACs, Edge Cases, Out of Scope, NFRs

**AC2 — templates/AGENTS.md upgraded**
- Given new project runs `npx create-agentic-pdlc`
- When scaffolded AGENTS.md is read by detailing agent
- Then same 7-section format is enforced

**AC3 — User story wrapper removed**
- Given current `templates/AGENTS.md` requires "As [user], I want..."
- When format is upgraded
- Then wrapper replaced by `## Problem` section in both files

## Edge Cases

- EC1: NFRs section absent on pure docs issue → allowed per explicit "omit only if..." note in format
- EC2: No baseline metric known → agent writes "TBD" not fabricated number
- EC3: Agent adds extra spec section not in format → "Omitting any section blocks stage:approval" rule keeps structure stable

## Out of Scope

- CI gate that validates spec quality — future Issue C, after adoption validated
- Changes to `docs/pdlc.md`
- Rewriting existing issue bodies

## Non-Functional Requirements

- Template purity: no `{{PLACEHOLDER}}` removed from `templates/AGENTS.md`
- Atomicity: both files updated in same PR — no partial state

## Files to Modify

- `AGENTS.md` — add `## Spec Format` section after `## Mandatory Workflow`
- `templates/AGENTS.md` — replace `### Spec format (Upstream Agents)` with 7-section format

### PR DESCRIPTION
**Pull request by rafaeltcosta86**
## Summary

- Adds `## Spec Format` section to root `AGENTS.md` with 7-section mandatory format
- Replaces user story wrapper (`As [user], I want...`) with Problem-first format in `templates/AGENTS.md`
- Updates `adapters/claude-code/skill.md` spec creation section with same format
- New mandatory sections: Problem, Sprint Goal/Metrics, Solution, ACs (Given/When/Then), Edge Cases, Out of Scope, NFRs

## Test plan

- [ ] Read root `AGENTS.md` — verify `## Spec Format` section present after `## Mandatory Workflow`
- [ ] Read `templates/AGENTS.md` — verify user story wrapper replaced by 7-section format
- [ ] Read `adapters/claude-code/skill.md` — verify Section 2 uses new format
- [ ] Confirm no `{{PLACEHOLDER}}` removed from `templates/AGENTS.md`

Closes #119

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -36,6 +36,43 @@ Always start from the current `main` HEAD. Never work over stale snapshots.
...
+- EC1: [condition] → [expected behavior]
+
+## Out of Scope
+- [item] — reason
+
+## Non-Functional Requirements
+- Performance: [metric with number]
+- Security: [constraint]
+- Reliability: [constraint]
+> Omit NFRs only for pure docs/markdown issues with zero runtime behavior.
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply AGENTS.md]: fixed.

File: templates/AGENTS.md [hunk 2]
```diff
@@ -39,29 +39,41 @@ Always start from the current `main` HEAD. Never work over stale snapshots.
 7. Run typecheck (if applicable): `{{TYPECHECK_COMMAND}}`
 8. Create a Pull Request with `Closes #N` in the body — automation moves the board.
 
-### Spec format (Upstream Agents)
+### Spec Format
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply templates/AGENTS.md]: Fixed.

File: templates/AGENTS.md [hunk 3]
```diff
@@ -39,29 +39,41 @@ Always start from the current `main` HEAD. Never work over stale snapshots.
...
+## Out of Scope
+- [item] — reason
 
-**AC2 — ...**
-...
+## Non-Functional Requirements
+- Performance: [metric with number]
+- Security: [constraint]
+- Reliability: [constraint]
+> Omit NFRs only for pure docs/markdown issues with zero runtime behavior.
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply templates/AGENTS.md]: Fixed.

File: adapters/claude-code/skill.md [hunk 4]
```diff
@@ -157,18 +157,39 @@ When asked to work on a feature, you will:
 - Stop and wait for the human PM's explicit approval (Gate 1).
 
 ### 2. Creating the Spec
-Once approved, you will detail the solution directly into the GitHub Issue body. Focus on precise Acceptance Criteria.
-**IMPORTANT:** You must always rewrite the full issue body to include both the user story and the Acceptance Criteria. Do not simply append the ACs to the existing text. Use this format:
+Once approved, detail the solution directly into the GitHub Issue body. Always rewrite the full issue body — never append only ACs to existing text. Use this format:
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply adapters/claude-code/skill.md]: Fixed.

File: adapters/claude-code/skill.md [hunk 5]
```diff
@@ -157,18 +157,39 @@ When asked to work on a feature, you will:
...
+- EC1: [condition] → [expected behavior]
+
+## Out of Scope
+- [item] — reason
+
+## Non-Functional Requirements
+- Performance: [metric with number]
+- Security: [constraint]
+- Reliability: [constraint]
+> Omit NFRs only for pure docs/markdown issues with zero runtime behavior.
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply adapters/claude-code/skill.md]: Fixed.
